### PR TITLE
[Integration Test] Generate worker artifact to AzureFunctionsPreRelease feed

### PR DESCRIPTION
**Pull request by Hazhzeng**
## Description

This opts-in Azure Functions Integration Test of Python Worker.
1. It allows dev branch to build Python Worker artifact because we want it to run integration test now.
2. Upload artifact to AzureFunctionsPreRelease for integration test in core tools when build from dev branch

## Test

Able to publish the artifact in AzureFunctionsPreRelease feed
image

/cc: @Francisco-Gamino 

---

### PR information

- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and CI is passing.

### Quality of Code and Contribution Guidelines
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-functions-python-worker/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -156,25 +157,36 @@ jobs:
...
       then
+        echo "Generating V3 Release Package for $BUILD_SOURCEBRANCHNAME"
         NUSPEC="pack\Microsoft.Azure.Functions.V3.PythonWorker.nuspec"
+        WKVERSION="$BUILD_SOURCEBRANCHNAME"
+      elif [[ $BUILD_SOURCEBRANCHNAME = dev ]]
+      then
+        echo "Generating V3 Integration Test Package for $BUILD_SOURCEBRANCHNAME"
+        LATEST_TAG=$(curl https://api.github.com/repos/Azure/azure-functions-python-worker/tags -s | jq '.[0].name' | sed 's/\"//g')
+        NUSPEC="pack\Microsoft.Azure.Functions.V3.PythonWorker.nuspec"
+        WKVERSION="3.$LATEST_TAG-dev"  # Integration Test (e.g. 3.1.1.6-dev)
```
Comment (vrdmr) [inline azure-pipelines.yml:183]: So - what this PR is doing is to greater a new nuspec package called `3.$LATEST_TAG-dev`, but not actually create a release on GitHub. Correct?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -191,3 +203,21 @@ jobs:
...
+    displayName: '[Integration Test] Set UploadPackage variable'
+  - task: NuGetCommand@2
+    condition: and(eq(variables.UploadPackage, true), eq(variables['Build.SourceBranch'], 'refs/heads/dev'))
+    inputs:
+      command: 'push'
+      packagesToPush: '$(Build.ArtifactStagingDirectory)/**/*.nupkg;!$(Build.ArtifactStagingDirectory)/**/*.symbols.nupkg'
+      nuGetFeedType: 'internal'
+      publishVstsFeed: 'e6a70c92-4128-439f-8012-382fe78d6396/f37f760c-aebd-443e-9714-ce725cd427df'
+      allowPackageConflicts: true
+    displayName: '[Integration Test] Push NuGet package to the AzureFunctionsPreRelease feed'
```
Comment (vrdmr) [inline azure-pipelines.yml:223]: nit;Please add a new line after this.

**Review by vrdmr — APPROVED** [review]
LGTM

File: azure-pipelines.yml (hunk 1 above)
Comment (Hazhzeng) [reply azure-pipelines.yml:183]: Yes, it does not create a release. Only an artifact in https://azfunc.visualstudio.com/Azure%20Functions/_packaging?_a=feed&feed=AzureFunctionsPreRelease

File: azure-pipelines.yml (hunk 2 above)
Comment (Hazhzeng) [reply azure-pipelines.yml]: Good catch!

**Comment by Francisco-Gamino** [comment]
Thanks for sending this out @Hazhzeng ! 

Question: This new tag `3.$LATEST_TAG-dev` will be the same as the one reference in the https://github.com/Azure/azure-functions-host/blob/dev/build/python.props#L3 and https://github.com/Azure/azure-functions-core-tools/blob/v3.x/src/Azure.Functions.Cli/Azure.Functions.Cli.csproj#L152 except that we are adding `-dev`, correct?

**Comment by Hazhzeng** [comment]
> Thanks for sending this out @Hazhzeng !
> 
> Question: This new tag `3.$LATEST_TAG-dev` will be the same as the one reference in the https://github.com/Azure/azure-functions-host/blob/dev/build/python.props#L3 and https://github.com/Azure/azure-functions-core-tools/blob/v3.x/src/Azure.Functions.Cli/Azure.Functions.Cli.csproj#L152 except that we are adding `-dev`, correct?

@Francisco-Gamino 
Yes, that is the same but we're adding -dev to it.
Basically, if the latest release version of Python worker is 1.1.12, then the package version will be 1.1.12-dev

**Pull request merged**